cppForSwig/Makefile

```diff
@@ -40,23 +42,33 @@ else
 PYTHON_INCLUDES=$(shell $(PYVER)-config --includes )
 endif
 
-CPPFLAGS += $(ARMORY_CPPFLAGS) -Icryptopp -Imdb -DUSE_CRYPTOPP -D__STDC_LIMIT_MACROS
-LDLIBS += -lpthread -Lmdb
-SWIG_OPTS    += -c++ -python -threads
-
-SWIG_INC     += 
-
-CXXCPP += $(CPPFLAGS) $(PYTHON_INCLUDES) -std=c++11
+# "-Wl,--no-as-needed" used for Ubuntu 12.04 due to rt library linking issues.
+# http://stackoverflow.com/questions/17150075/undefined-reference-to-clock-gettime-although-lrt-is-given
+# has more info. Require only for static linking. When 12.04 goes, remove this
+# unless it's discovered that OS X needs it.
+ifdef STATIC_LINK
+LDLIBS += -Wl,--no-as-needed
```

Comment:
I just tried moving -lrt to the very end of LDLIBS on https://github.com/josephbisch/BitcoinArmory/tree/test/. It doesn't work.

I tried building on 14.04 (using Gitian, though that's not significant). Then I tried running the result in my 12.04 chroot. I get the following error:

```
root@crunchbang:~/1# python ArmoryQt.py 
Traceback (most recent call last):
  File "ArmoryQt.py", line 39, in <module>
    import CppBlockUtils as Cpp
  File "/root/1/CppBlockUtils.py", line 28, in <module>
    _CppBlockUtils = swig_import_helper()
  File "/root/1/CppBlockUtils.py", line 24, in swig_import_helper
    _mod = imp.load_module('_CppBlockUtils', fp, pathname, description)
ImportError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.17' not found (required by /root/1/_CppBlockUtils.so)
```

The output of ldd:

```
root@crunchbang:~/1# ldd _CppBlockUtils.so 
./_CppBlockUtils.so: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.17' not found (required by ./_CppBlockUtils.so)
	linux-vdso.so.1 =>  (0x00007ffc50530000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f6a7cdd0000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f6a7cad0000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f6a7c710000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f6a7d890000)
```

So librt doesn't appear to be linked when building under Ubuntu without --no-as-needed. That appears to be irrespective of the position of -lrt.

Is there something better than --no-as-needed to use?